doc/mgmt/OpenConfig_Interfaces.md

```diff
@@ -0,0 +1,199 @@
+# OpenConfig support for Ethernet interfaces.
+
+# High Level Design Document
+#### Rev 0.1
+
+# Table of Contents
+  * [List of Tables](#list-of-tables)
+  * [Revision](#revision)
+  * [About This Manual](#about-this-manual)
+  * [Scope](#scope)
+  * [Definition/Abbreviation](#definitionabbreviation)
+  * [1 Feature Overview](#1-feature-overview)
+    * [1.1 Requirements](#11-requirements)
+      * [1.1.1 Functional Requirements](#111-functional-requirements)
+      * [1.1.2 Configuration and Management Requirements](#112-configuration-and-management-requirements)
+      * [1.1.3 Scalability Requirements](#113-scalability-requirements)
+    * [1.2 Design Overview](#12-design-overview)
+      * [1.2.1 Basic Approach](#121-basic-approach)
+      * [1.2.2 Container](#122-container)
+  * [2 Functionality](#2-functionality)
+      * [2.1 Target Deployment Use Cases](#21-target-deployment-use-cases)
+  * [3 Design](#3-design)
+    * [3.1 Overview](#31-overview)
+    * [3.2 DB Changes](#32-db-changes)
+      * [3.2.1 CONFIG DB](#321-config-db)
+      * [3.2.2 APP DB](#322-app-db)
+      * [3.2.3 STATE DB](#323-state-db)
+      * [3.2.4 ASIC DB](#324-asic-db)
+      * [3.2.5 COUNTER DB](#325-counter-db)
+    * [3.3 CLI](#36-cli)
+      * [3.3.1 Data Models](#361-data-models)
+      * [3.3.2 REST API Support](#363-rest-api-support)
+      * [3.3.3 gNMI Support](#364-gnmi-support)
+  * [4 Flow Diagrams](#4-flow-diagrams)
+  * [5 Error Handling](#5-error-handling)
+  * [6 Unit Test Cases](#9-unit-test-cases)
+    * [6.1 Functional Test Cases](#91-functional-test-cases)
+    * [6.2 Negative Test Cases](#92-negative-test-cases)
+  
+# List of Tables
+[Table 1: Abbreviations](#table-1-abbreviations)
+
+# Revision
+| Rev |     Date    |       Author          | Change Description                |
+|:---:|:-----------:|:---------------------:|-----------------------------------|
+| 0.1 | 02/24/2024  | Nikita Agarwal / Satoru Shinohara | Initial version                   |
+
+# About this Manual
+This document provides general information about the OpenConfig configuration of Ethernet interfaces in SONiC.
+
+# Scope
+- This document describes the high level design of OpenConfig configuration of Ethernet interfaces via REST and gNMI in SONiC.
+- This does not cover the SONiC CLICK and KLISH CLI.
+- This covers only Ethernet interfaces configuration.
+- This does not support subinterfaces configuration.
+- This does not cover gNMI subscription.
+- This does not cover secondary IP configuration.
+- Alias support for the interface names are not available to configure for Ethernet.
+

```

Comment:
Let's clarify the scope here, by presenting the OC-yang tree for the supported attributes as part of this HLD. Community should not expect all openconfig ethernet configs are supported in this HLD.

Reply:
I've updated the OC Yang tree and addressed all the review comments as well.